myshop/tests/cart/test_cart.py

```diff
@@ -0,0 +1,130 @@
+from django.test import TestCase, RequestFactory
+from tests.shop.test_modelmixintestcase import ModelMixinTestCase
+from cart.cart import Cart
+from django.conf import settings
+from django.contrib.sessions.middleware import SessionMiddleware
+from shop.models import Product, Category
+
+
+class Testurls(TestCase):
+    def setUp(self):
+        self.request = RequestFactory().get("/")
+        middleware = SessionMiddleware(get_response=self.request)
+        middleware.process_request(self.request)
+        self.request.session.save()
+
+        self.cart_obj = Cart(self.request)
+
+        self.category = Category.objects.create(
+            name="Gaming",
+            slug="gaming",
+        )
+
+        self.product = Product.objects.create(
+            category=self.category,
+            name="Play station 5",
+            slug="play-station-5",
+            description="A gaming console",
+            price=50000.00,
+        )
+
+        self.product_id_str = str(self.product.id)
+
+    def test_initialize_cart_clean_session(self):
+        empty_cart = {}
+        self.assertEqual(empty_cart, self.cart_obj.cart)
+
+    def test_add_new_product_to_the_cart(self):
+        quantity = 21
+        self.cart_obj.add_product(self.product, quantity=quantity)
+        cart = self.request.session.get(settings.CART_SESSION_ID)
+
+        self.assertEqual(cart[self.product_id_str]["quantity"], quantity)
+        self.assertEqual(
+            str(self.product.price), cart[self.product_id_str]["price"]
+        )
+
+    def test_add_new_product_to_the_cart_without_quantity_arg_sets_default_quantity(
```

Comment:
```suggestion
    def test_adding_product_to_the_cart_without_quantity_sets_default_quantity(
```

args is adding too much programming detail and making the name less readable